standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/SharedCache.java

```diff
@@ -1434,6 +1699,35 @@ public boolean populateTableInCache(Table table, ColumnStatistics tableColStats,
     tblWrapper.isTableColStatsCacheDirty.set(false);
     tblWrapper.isPartitionColStatsCacheDirty.set(false);
     tblWrapper.isAggrPartitionColStatsCacheDirty.set(false);
+
+    if (cacheObjects.getPrimaryKeys() != null) {
+      if(!tblWrapper.cachePrimaryKeys(cacheObjects.getPrimaryKeys(), true)) {
+        return false;
+      }
+    }
+    tblWrapper.memberCacheDirty[MemberName.PRIMARY_KEY_CACHE.ordinal()].set(false);
```

Comment:
TableWrapper can have setMemberCacheDirty(MemberName, boolean) method to make it clean.